sqlglot/schema.py

```diff
@@ -1,13 +1,24 @@
 import abc
+from copy import copy
 
-from sqlglot import exp
+from sqlglot import expressions as exp
 from sqlglot.errors import OptimizeError
-from sqlglot.helper import csv_reader
+from sqlglot.helper import csv_reader, ensure_table
 
 
 class Schema(abc.ABC):
     """Abstract base class for database schemas"""
 
+    @abc.abstractmethod
```

Comment:
Adding an abstractmethod makes this a backwards incompatible change.

I also think of `Schema` as being owned by the optimizer - i.e., it defines the interface that the optimizer depends on.

I think we want a second interface - `MutableSchema`.

Reply:
does this need to be an abc?

Reply:
@barakalon Yeah I think the second `MutableSchema` interface makes sense. Working on that change now.